codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketProtocolHandler.java

```diff
@@ -48,9 +48,15 @@ protected void decode(ChannelHandlerContext ctx, WebSocketFrame frame, List<Obje
         if (frame instanceof PingWebSocketFrame) {
             frame.content().retain();
             ctx.channel().writeAndFlush(new PongWebSocketFrame(frame.content()));
+            if (!ctx.channel().config().isAutoRead()) {
+                ctx.read();
+            }
```

Comment:
nit you could factor out this into a method and then just call the method:

```
private static void readIfNeeded(ChannelHandlerContext ctx) {
    if (!ctx.channel.config().isAutoRead()) {
        ctx.read();
    }
}
```

Reply:
done